src/charm.py

```diff
@@ -1496,28 +1549,36 @@ def get_resources_limits(self, container_name: str) -> Dict:
         return {}
 
     def get_node_allocable_memory(self) -> int:
-        """Return the allocable memory in bytes for a given node.
-
-        Args:
-            node_name: name of the node to get the allocable memory for
-        """
+        """Return the allocable memory in bytes for the current K8S node."""
         client = Client()
         node = client.get(Node, name=self._get_node_name_for_pod(), namespace=self._namespace)
         return any_memory_to_bytes(node.status.allocatable["memory"])
 
-    def get_available_memory(self) -> int:
-        """Get available memory for the container in bytes."""
+    def get_node_cpu_cores(self) -> int:
+        """Return the number of CPU cores for the current K8S node."""
+        client = Client()
+        node = client.get(Node, name=self._get_node_name_for_pod(), namespace=self._namespace)
+        return int(node.status.allocatable["cpu"])
+
+    def get_available_resources(self) -> Tuple[int, int]:
+        """Get available CPU cores and memory (in bytes) for the container."""
+        cpu_cores = self.get_node_cpu_cores()
         allocable_memory = self.get_node_allocable_memory()
         container_limits = self.get_resources_limits(container_name="postgresql")
+        if "cpu" in container_limits:
+            cpu_str = container_limits["cpu"]
+            constrained_cpu = any_memory_to_bytes(cpu_str)
+            if constrained_cpu < allocable_memory:
+                logger.debug(f"CPU constrained to {cpu_str} cores from resource limit")
+                cpu_cores = constrained_cpu
```

Comment:
Any comments why do we compare CPUs to Memory?

Reply:
It's a mistake. Good catch. I'll fix it.

Reply:
Fixed on https://github.com/canonical/postgresql-k8s-operator/pull/281/commits/72477c847ab3f5bb1c64edd1b6bfac50243e9787.